ci/kind/kind-install.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+LATEST_KIND_VERSION=$(head -n1 ./ci/kind/version)
+
+# on linux
+function upgrade_kind {
+    arch_name="$(uname -m)"
+    binary_name=""
+    case "$arch_name" in
+     "x86_64")
+      binary_name="kind-linux-amd64"
+      ;;
+      "aarch64")
+      binary_name="kind-linux-arm64"
+      ;;
+      *)
+      echoerr "Unsupported platform $arch_name"
+      exit 1
+      ;;
+    esac
+
+    set +e
+    curl -Lo ./kind https://kind.sigs.k8s.io/dl/$LATEST_KIND_VERSION/$binary_name
+    chmod +x ./kind
+    sudo mv kind /usr/local/bin
+    if [[ $? -ne 0 ]]; then
+        echoerr "Failed to upgrade Kind version $LATEST_KIND_VERSION"
+        exit 1
+    else
+        echo "Kind version $LATEST_KIND_VERSION upgraded successfully"   
+    fi
+    set -e
+}
+
+CMD_RETURN_CODE=0
+kind_version=$(kind version | awk  '{print $2}') || CMD_RETURN_CODE=$? || true
```

Comment:
the `|| true` should not be required, the `CMD_RETURN_CODE=$?` will swallow the possible error code